src/webnn/native/dml/GraphDML.cpp

```diff
@@ -316,15 +316,18 @@ namespace webnn::native::dml {
                     input = ::dml::Reinterpret(input, newInputDims,
                                                CalculateInputLayoutStrides(NhwcToNchw, inputDims));
                     break;
-                case NchwToNhwc:
+                case NchwToNhwc: {
                     newInputDims.resize(4);
                     newInputDims[0] = inputDims[0];
                     newInputDims[1] = inputDims[2];
                     newInputDims[2] = inputDims[3];
                     newInputDims[3] = inputDims[1];
-                    input = ::dml::Reinterpret(input, newInputDims,
-                                               CalculateInputLayoutStrides(NchwToNhwc, inputDims));
-                    break;
+                    ::dml::Optional<::dml::TensorStrides> strides = input.GetOutputDesc().strides;
+                    DAWN_ASSERT(strides != ::dml::NullOpt);
+                    ::dml::TensorStrides newStrides = {strides.value()[0], strides.value()[2],
+                                                       strides.value()[3], strides.value()[1]};
+                    input = ::dml::Reinterpret(input, newInputDims, newStrides);
```

Comment:
Why doesnot `NhwcToNchw` case use the `TensorStrides`?

Reply:
I think the strides will be computed by default in nchw layout, so you can move the content of CalculateInputLayoutStrides here, and remove the `CalculateInputLayoutStrides` function, does it make sense?

Reply:
> I think the strides will be computed by default in nchw layout, so you can move the content of CalculateInputLayoutStrides here, and remove the `CalculateInputLayoutStrides` function, does it make sense?

::dml::Reinterpret is not a real operator, it will not calculate anything, we need set strides and dimensions into it.@fujunwei

Reply:
I mean you can alos use `TensorStrides ` instead of `CalculateInputLayoutStrides `, we don't need to mantain two approaches to convert layout.

Reply:
I know what you mean, but the tensorStrides may be null, not always have value, for example, you run a test case with a nhwc input tensor. so we need calculate them explicitly here. @fujunwei

Reply:
Yes, please calcuate it explicitly  with moving the code from CalculateInputLayoutStrides.

Reply:
I removed the calculation here, PTAL, thanks! @fujunwei